pkg/defaults/components.go

```
@@ -0,0 +1,78 @@
+}
+
+func NewDefaultsComponent(config *kuma_cp.Defaults, cpMode config_core.CpMode, environment config_core.EnvironmentType, resManager core_manager.ResourceManager, resStore store.ResourceStore) component.Component {
+	return &defaultsComponent{
+		cpMode:      cpMode,
+		environment: environment,
+		config:      config,
+		resManager:  resManager,
+		resStore:    resStore,
+	}
+}
+
+var _ component.Component = &defaultsComponent{}
+
+type defaultsComponent struct {
+	cpMode      config_core.CpMode
+	environment config_core.EnvironmentType
+	config      *kuma_cp.Defaults
+	resManager  core_manager.ResourceManager
+	resStore    store.ResourceStore
+}
+
+func (d *defaultsComponent) NeedLeaderElection() bool {
+	// If you spin many instances without default resources at once, many of them would create them, therefore only leader should create default resources.
+	return true
+}
+
+func (d *defaultsComponent) Start(_ <-chan struct{}) error {
+	// todo(jakubdyszkiewicz) once this https://github.com/kumahq/kuma/issues/1001 is done. Wait for all the components to be ready.
+	if d.config.SkipMeshCreation {
+		log.V(1).Info("skipping default Mesh creation because KUMA_DEFAULTS_SKIP_MESH_CREATION is set to true")
+	} else if err := retryOperation(d.createMeshIfNotExist); err != nil {
+		// Retry this operation since on Kubernetes Mesh needs to be validated and set default values.
+		// This code can execute before the control plane is ready therefore hooks can fail.
+		return errors.Wrap(err, "could not create the default Mesh")
+	}
+
+	if !d.shouldCreateSigningKey() {
+		log.V(1).Info("skip creating default Signing Key since CP with this mode is not required to have a key", "env", d.environment, "mode", d.cpMode)
+	} else if err := d.createSigningKeyIfNotExist(); err != nil {
+		return errors.Wrap(err, "could not create the default Signing Key")
+	}
+	return nil
+}
+
+func retryOperation(fn func() error) error {
+	backoff, err := retry.NewConstant(1 * time.Second)
```

Can we just ignore the error since constant `1 * time.Second` is hardcoded?

---

I'm anxious about ignoring errors in Go... They may change something in the library and add extra check, we update this and error is ignore. But in this case I hope it does not happen - changed